pkg/auth/roles_authz.go

```diff
@@ -0,0 +1,95 @@
+package auth
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/environments"
+	shared "github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/shared"
+	arrayUtils "github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/shared/utils/arrays"
+	"github.com/pkg/errors"
+	"github.com/spf13/pflag"
+	"gopkg.in/yaml.v2"
+)
+
+var _ environments.ConfigModule = (*AdminRoleAuthZConfig)(nil)
+
+// RolesConfiguration is the configuration of required roles per HTTP method of the admin API.
+type RolesConfiguration struct {
+	HTTPMethod string   `yaml:"method"`
+	RoleNames  []string `yaml:"roles"`
+}
+
+// RoleConfig represents the role configuration.
+type RoleConfig []RolesConfiguration
+
+// AdminRoleAuthZConfig is the configuration of the role authZ middleware.
+type AdminRoleAuthZConfig struct {
+	Enabled         bool
+	RolesConfigFile string
+	RolesConfig     RoleConfig
+}
+
+// NewAdminAuthZConfig creates a default AdminRoleAuthZConfig which is enabled and uses the production configuration.
+func NewAdminAuthZConfig() *AdminRoleAuthZConfig {
+	return &AdminRoleAuthZConfig{
+		Enabled:         true,
+		RolesConfigFile: "config/admin-authz-configuration.yaml",
+	}
+}
+
+// AddFlags adds required flags for the role authZ configuration.
+func (c *AdminRoleAuthZConfig) AddFlags(fs *pflag.FlagSet) {
+	fs.StringVar(&c.RolesConfigFile, "admin-authz-config-file", c.RolesConfigFile,
+		"Admin API authZ configuration file containing list of required role per API method")
+	fs.BoolVar(&c.Enabled, "enable-admin-authz", c.Enabled, "Enable admin API authZ via roles")
+}
+
+// ReadFiles will read and validate the contents of the configuration file.
+func (c *AdminRoleAuthZConfig) ReadFiles() error {
+	if c.Enabled {
+		if err := readRoleAuthZConfigFile(c.RolesConfigFile, &c.RolesConfig); err != nil {
+			return err
+		}
+		return validateRolesConfiguration(c.RolesConfig)
+	}
+	return nil
+}
+
```

Comment:
```suggestion
func (c *AdminRoleAuthZConfig) ReadFiles() error {
	if c.Enabled {
		if err := readRoleAuthZConfigFile(c.RolesConfigFile, &c.RolesConfig); err != nil {
			return err
		}
	}
	return nil
}

```

This method does two things:
- it reads the file 
- it also perform the validation

It is possible to remove the validation method into a separate method and make the config as a `ServiceValidator()`

in https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/fd1700778497d649ae4943d8d732c664f6361e28/pkg/providers/core.go#L42
it will read 
```go

di.Provide(auth.NewAdminAuthZConfig, di.As(new(environments.ConfigModule)), di.As(new(environments.ServiceValidator))),
```

And to perform the validation, define a `Validate` method for the `AdminRoleAuthZConfig`  config with the below signature
```go
func (c *AdminRoleAuthZConfig) Validate(env *environments.Env) error {
	if c.Enabled {
		if err := readRoleAuthZConfigFile(c.RolesConfigFile, &c.RolesConfig); err != nil {
			return err
		}
		return validateRolesConfiguration(c.RolesConfig)
	}
	return nil
}
```